### DISCUSSION TITLE
How to use WriteStream with api endpoints?

### DISCUSSION THREAD

**Discussion by AhsanSarwar45** [sveltejs/kit#10249] [Q&A]
I have a 'stream' function that returns a WriteStream. The usage in express is like this:
```js
const stream = require('youtube-audio-stream');

const uri = "https://www.youtube.com/watch?v=t1TcDHrkQYg";

app.get("/stream", (req, res) => {
  stream(uri).pipe(res);
})
```

How can I do the equivalent in svelte kit api endpoints? The docs gives example of ReadableStream but not WriteStream.

**Comment by karimfromjordan** [comment]
If your package only works with Node.js streams and not web streams you likely have to convert between the two using `Readable/Writable.fromWeb()/toWeb()`. The `Response` constructor accepts `ReadableStream`s as the body argument. Make sure to read about `Readable.toWeb()` in the Node.js docs and GitHub issues because it can consume too much memory if you don't limit it.

```js
// +server.js
import { Readable } from 'node:stream';
import stream from 'youtube-audio-stream';

export function POST(event) {
  const web_rstream = Readable.toWeb(stream(url));
  
  return new Response(web_rstream);
}
```

It requires at least Node.js 17.0.0.
